File: codeforces/323div2/583A.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
	int N,i,j,hi,vi,row[501]={0},col[501]={0},count=0;
	cin>>N;
	for(i=0;i<N*N;i++){
		cin>>hi>>vi;count++;
		if(row[hi]==0&&col[vi]==0){
			cout<<count<<" ";row[hi]=1;col[vi]=1;}
	}
	return 0;
	}

File: codeforces/323div2/583A.py
N = int(raw_input())
row = [0]*(N+1)
col = [0]*(N+1)
i = 0
str1 = ""
while i<N*N:
	lol = map(int,raw_input().split())
	if row[lol[0]]==0 and col[lol[1]]==0:
		str1+=str(i+1)+" "
		row[lol[1]]=1
		col[lol[1]]=1
	i+=1
print str1
